File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/logic/model/book/google/GoogleBookData.java
package ru.kmorozov.gbd.core.logic.model.book.google;

import com.google.gson.annotations.SerializedName;
import ru.kmorozov.gbd.core.logic.model.book.base.IBookData;

import java.io.Serializable;
import java.net.URI;

import static ru.kmorozov.gbd.core.logic.extractors.google.GoogleImageExtractor.BOOK_ID_PLACEHOLDER;
import static ru.kmorozov.gbd.core.logic.extractors.google.GoogleImageExtractor.HTTPS_TEMPLATE;

/**
 * Created by km on 28.11.2015.
 */
public class GoogleBookData implements IBookData, Serializable {

    @SerializedName("fullview")
    private boolean fullview;
    @SerializedName("page_width")
    private int pageWidth;
    @SerializedName("page_height")
    private int pageHeight;
    @SerializedName("font_height")
    private int fontHeight;
    @SerializedName("first_content_page")
    private int firstContentPage;
    @SerializedName("disable_twopage")
    private boolean disableTwopage;
    @SerializedName("initial_zoom_width_override")
    private int initialZoomWidthOverride;
    @SerializedName("show_print_pages_button")
    private boolean showPrintPagesButton;
    @SerializedName("title")
    private String title;
    @SerializedName("subtitle")
    private String subtitle;
    @SerializedName("attribution")
    private String attribution;
    @SerializedName("additional_info")
    private AdditionalInfo additionalInfo;
    @SerializedName("table_of_contents_page_id")
    private String tableOfContentsPageId;
    @SerializedName("max_resolution_image_width")
    private int maxResolutionImageWidth;
    @SerializedName("max_resolution_image_height")
    private int maxResolutionImageHeight;
    @SerializedName("num_toc_pages")
    private int numTocPages;
    @SerializedName("quality_info")
    private String qualityInfo;
    @SerializedName("volume_id")
    private String volumeId;
    @SerializedName("permission_info")
    private String permissionInfo;
    @SerializedName("is_ebook")
    private boolean ebook;
    @SerializedName("volumeresult")
    private VolumeResultFlags volumeresult;
    @SerializedName("rating")
    private int rating;
    @SerializedName("num_reviews")
    private int numReviews;
    @SerializedName("publisher")
    private String publisher;
    @SerializedName("publication_date")
    private String publicationDate;
    @SerializedName("num_pages")
    private int numPages;
    @SerializedName("sample_url")
    private String sampleUrl;
    @SerializedName("synposis")
    private String synposis;
    @SerializedName("my_library_url")
    private String myLibraryUrl;
    @SerializedName("is_magazine")
    private boolean magazine;
    @SerializedName("is_public_domain")
    private boolean publicDomain;
    @SerializedName("last_page")
    private GooglePageInfo lastPage;

    @Override
    public String getTitle() {
        return title;
    }

    @Override
    public String getVolumeId() {
        return volumeId;
    }

    public VolumeResultFlags getFlags() {
        return volumeresult;
    }

    public URI getBaseUrl() {
        return URI.create(HTTPS_TEMPLATE.replace(BOOK_ID_PLACEHOLDER, volumeId));
    }
}

File: ru.kmorozov.gbd.db/src/main/java/ru/kmorozov/db/core/logic/model/book/google/GoogleBookData.kt
package ru.kmorozov.db.core.logic.model.book.google

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import ru.kmorozov.gbd.core.logic.model.book.base.IBookData

import java.io.Serializable
import java.net.URI

import ru.kmorozov.gbd.core.config.constants.GoogleConstants.BOOK_ID_PLACEHOLDER
import ru.kmorozov.gbd.core.config.constants.GoogleConstants.HTTPS_TEMPLATE

/**
 * Created by km on 28.11.2015.
 */
class GoogleBookData : IBookData, Serializable {

    @Expose @SerializedName("fullview")
    private val fullview: Boolean = false
    @Expose @SerializedName("page_width")
    private val pageWidth: Int = 0
    @Expose @SerializedName("page_height")
    private val pageHeight: Int = 0
    @Expose @SerializedName("font_height")
    private val fontHeight: Int = 0
    @Expose @SerializedName("first_content_page")
    private val firstContentPage: Int = 0
    @Expose @SerializedName("disable_twopage")
    private val disableTwopage: Boolean = false
    @Expose @SerializedName("initial_zoom_width_override")
    private val initialZoomWidthOverride: Int = 0
    @Expose @SerializedName("show_print_pages_button")
    private val showPrintPagesButton: Boolean = false
    @Expose @SerializedName("title")
    override lateinit var title: String
    @Expose @SerializedName("subtitle")
    private val subtitle: String? = null
    @Expose @SerializedName("attribution")
    private val attribution: String? = null
    @Expose @SerializedName("additional_info")
    val additionalInfo: AdditionalInfo? = null
    @Expose @SerializedName("table_of_contents_page_id")
    val tableOfContentsPageId: String? = null
    @Expose @SerializedName("max_resolution_image_width")
    private val maxResolutionImageWidth: Int = 0
    @Expose @SerializedName("max_resolution_image_height")
    private val maxResolutionImageHeight: Int = 0
    @Expose @SerializedName("num_toc_pages")
    private val numTocPages: Int = 0
    @Expose @SerializedName("quality_info")
    private val qualityInfo: String? = null
    @Expose @SerializedName("volume_id")
    override lateinit var volumeId: String
    @Expose @SerializedName("permission_info")
    private val permissionInfo: String? = null
    @Expose @SerializedName("is_ebook")
    private val ebook: Boolean = false
    @Expose @SerializedName("volumeresult")
    val flags: VolumeResultFlags? = null
    @Expose @SerializedName("rating")
    private val rating: Int = 0
    @Expose @SerializedName("num_reviews")
    private val numReviews: Int = 0
    @Expose @SerializedName("publisher")
    val publisher: String? = null
    @Expose @SerializedName("publication_date")
    val publicationDate: String? = null
    @Expose @SerializedName("num_pages")
    val numPages: Int = 0
    @Expose @SerializedName("sample_url")
    private val sampleUrl: String? = null
    @Expose @SerializedName("synposis")
    private val synposis: String? = null
    @Expose @SerializedName("my_library_url")
    private val myLibraryUrl: String? = null
    @Expose @SerializedName("is_magazine")
    private val magazine: Boolean = false
    @Expose @SerializedName("is_public_domain")
    private val publicDomain: Boolean = false
    @Expose @SerializedName("last_page")
    private val lastPage: GooglePageInfo? = null

    val baseUrl: URI
        get() = URI.create(HTTPS_TEMPLATE.replace(BOOK_ID_PLACEHOLDER, volumeId))

}
